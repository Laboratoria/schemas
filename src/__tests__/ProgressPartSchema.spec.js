const mongoose = require('mongoose/browser');
const {
  ProgressPartSchema, TopicUnitPartSchema,
} = require('../..')(mongoose);

describe('ProgressPartSchema', () => {
  it('should fail validation when fields missing', () => {
    const doc = new mongoose.Document({}, ProgressPartSchema);
    return doc.validate()
      .catch(err => expect(err.errors).toMatchSnapshot());
  });

  it('should pass at least with unit field', () => {
    const topicUnitPart = new mongoose.Document({}, TopicUnitPartSchema);
    const doc = new mongoose.Document({
      part: topicUnitPart._id,
    }, ProgressPartSchema);

    return doc.validate();
  });

  it('should pass with all fields', () => {
    const topicUnitPart = new mongoose.Document({}, TopicUnitPartSchema);
    const doc = new mongoose.Document({
      part: topicUnitPart._id,
      openedAt: Date.now(),
      completedAt: Date.now(),
      latestActivity: true,
    }, ProgressPartSchema);

    return doc.validate();
  });
});
