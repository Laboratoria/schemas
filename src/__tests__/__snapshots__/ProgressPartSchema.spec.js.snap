// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressPartSchema should fail validation when fields missing 1`] = `
Object {
  "part": [ValidatorError: Path \`part\` is required.],
}
`;
